### DISCUSSION TITLE
AIR and POS integration

### DISCUSSION THREAD

**Discussion by radubirsan** [airsdk/Adobe-Runtime-Support#2121] [Q&A]
I am looking for examples of integrating a Cash / Credit Card reader and Adobe AIR.  People will pay adobe AIR prints a ticket once payment is confirmed. Is there any native extension that helps me to do this. ( Preferably on windows )

**Comment by raresn** [comment]
Very interested in this as well!

**Comment by Dextercitox** [comment]
I have made some of this kind of integrations. Actually the usual way to work is to have the API or SDK of the payment system or bank to send the operation data ( total amount, invoice_id ... ) to the card reader and once the payment is made, you receive the callback of the device to end the transaction and print the receipt.

The payment process is made in the device with the bank/payment system app and you dont have to deal with any security measures. Handling payments it's something dangerous and in most countries you need to be an authorized payment processor to be able to do that, and that's complicated and expensive.

So the approach is to find the right payment platform and check what they have available to make the integration. Some of them uses IP socket connections, other some cloud API and other some native SDK ( not valid in windows obviusly ).

Hope it helps !

:)

**Reply by radubirsan** [reply]
"Some of them uses IP socket connections" can I use AIR Socket connection to communicate with the payment device with something like the MDB Protocol?

**Reply by Dextercitox** [reply]
Yes it's possible, but the payment device API has to support it.
